src/EditorFeatures/Core.Wpf/InlineRename/AbstractEditorInlineRenameService.SymbolRenameInfo.cs

```diff
@@ -190,45 +186,13 @@ public string GetFinalSymbolName(string replacementText)
                 return replacementText;
             }
 
-            public Task<IInlineRenameLocationSet> FindRenameLocationsAsync(OptionSet? optionSet, CancellationToken cancellationToken)
-            {
-                Task<RenameLocations> renameTask;
-                lock (_gate)
-                {
-                    if (_underlyingFindRenameLocationsTask == null)
-                    {
-                        // If this is the first call, then just start finding the initial set of rename
-                        // locations.
-                        var solution = _document.Project.Solution;
-                        _underlyingFindRenameLocationsTask = Renamer.FindRenameLocationsAsync(
-                            solution, this.RenameSymbol, RenameOptionSet.From(solution, optionSet), cancellationToken);
-                        renameTask = _underlyingFindRenameLocationsTask;
-
-                        // null out the option set.  We don't need it anymore, and this will ensure
-                        // we don't call FindWithUpdatedOptionsAsync below.
-                        optionSet = null;
-                    }
-                    else
-                    {
-                        // We already have a task to figure out the set of rename locations.  Let it
-                        // finish, then ask it to get the rename locations with the updated options.
-                        renameTask = _underlyingFindRenameLocationsTask;
-                    }
-                }
-
-                return GetLocationSetAsync(renameTask, optionSet, cancellationToken);
-            }
-
-            private async Task<IInlineRenameLocationSet> GetLocationSetAsync(Task<RenameLocations> renameTask, OptionSet? optionSet, CancellationToken cancellationToken)
+            public async Task<IInlineRenameLocationSet> FindRenameLocationsAsync(OptionSet? optionSet, CancellationToken cancellationToken)
             {
-                var locationSet = await renameTask.ConfigureAwait(false);
-                if (optionSet != null)
-                {
-                    locationSet = await locationSet.FindWithUpdatedOptionsAsync(
-                        RenameOptionSet.From(_document.Project.Solution, optionSet), cancellationToken).ConfigureAwait(false);
```

Comment:
this only allows some amount of reuse.  but it's reuse in an area that is already aggressively cached in OOP, so it doesnt' seem to actually buy anything from manual testing.  also, it only caches the symbolic part of hte operation.  if someone enables 'overloads' then enables 'comments, we'll redo both parts of those additional searches both times.  i.e. we don't cache the *further* parts of the operations we're performing.  only whatever was computed in teh initial 'find locations' task.